### PR TITLE
Advanced Editor from Inline Editor

### PR DESCRIPTION
**Pull request by kappu72**
## Description
refer to geosolutions-it/mapstore2-c039/issues/122
## Issues
 - #4563 

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [X] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
It's not possible to open the advanced map editor from the inline editor toolbar.

**What is the new behavior?**
It's now possible to launch the advanced map editor plugin from the inline map editor.
On cancel the user ii brought back to online editor, on save, the map is saved inside the geostory content as it is and the inline editor is closed

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/translations/data.en-US [hunk 1]
```diff
@@ -1939,7 +1939,8 @@
                 "saveChanges": "Save changes",
                 "closeMapEditing": "Close",
                 "confirmCloseMapEditing": "Close Map Editing",
-                "pendingChangesDiscardConfirm": "Discard current map changes?"
+                "pendingChangesDiscardConfirm": "Discard current map changes?",
+                "advancedMapEditor": "Open advanced map editor"
```
[the remark this answers is not in the document]
Comment (kappu72) [reply web/client/translations/data.en-US]: It's no more original map, as requested by @tdipisa, a user edits the content map.

File: web/client/components/geostory/common/enhancers/map.jsx [hunk 2]
```diff
@@ -84,15 +97,19 @@ const ResetButton = (props) => (<ConfirmButton
...
             glyph: "floppy-disk",
-            visible: true,
             disabled: !pendingChanges,
             tooltipId: "geostory.contentToolbar.saveChanges",
             onClick: toggleEditing
         }, {
             renderButton: <ResetButton disabled={disableReset} onClick={onReset}/>
+        },
+        {
+            glyph: "pencil",
```
[the remark this answers is not in the document]
Comment (kappu72) [reply web/client/components/geostory/common/enhancers/map.jsx]: yes I'll add

**Comment by kappu72** [comment]
> Hi, please see my comments.
> 
> i noticed that when you enter the advanced editor from inline editor, and you click cancel or the X
> then the confirm modal it is shown even if no changes have been done from the user
On exit without saving it always asks confirmation to the user. It's almost impossible to understand when a map has been edited or not. Consider that when the map is drown a map view change action is dispatched by the map component to fit the map in the map container width and the center it's almost always a little bit different to the original one

File: web/client/epics/__tests__/geostory-test.js [hunk 3]
```diff
@@ -1072,4 +1080,38 @@ describe('Geostory Epics', () => {
...
+    it('inlineEditorEditMap on advancedEditing save', done => {
+        const launchActions = [mapEditorShow("inlineEditor", {data: {}, id: 1}), saveMapEditor({}, "inlineEditor")];
+
+        const NUM_ACTIONS = 3;
+        testEpic(inlineEditorEditMap, NUM_ACTIONS, launchActions, (actions) => {
+            expect(actions.length).toBe(NUM_ACTIONS);
+            actions.forEach(({type, ...a}) => {
+                switch (type) {
+                case UPDATE:
+                    a.path.endsWith(".map") || a.path.endsWith(".editMap");
```
[the remark this answers is not in the document]
Comment (kappu72) [reply web/client/epics/__tests__/geostory-test.js]: yes

**Comment by MV88** [comment]
> > i noticed that when you enter the advanced editor from inline editor, and you click cancel or the X
> > then the confirm modal it is shown even if no changes have been done from the user
> 
> On exit without saving it always asks confirmation to the user. It's almost impossible to understand when a map has been edited or not. Consider that when the map is drown a map view change action is dispatched by the map component to fit the map in the map container width and the center it's almost always a little bit different to the original one

understood

**Pull request merged**